[Bug?]: TS error in `preview.tsx` & `main.ts` when using absolute (`src/`-prefixed) and/or component directory imports [storybook-vite]

### What's not working?

As reported [in the forum](https://community.redwoodjs.com/t/storybook-in-redwood-is-moving-to-vite/7212/21?u=philzen) during the storybook migration from webpack to vite:

![screenshot](https://rwjs-discourse.nyc3.cdn.digitaloceanspaces.com/original/2X/7/72b4d4d9ef6660d978b110bda60acf5a0b28fc09.png)

Above squiggles amount to `Cannot find module '{import path}' or its corresponding type declarations.`. Storybook compiles fine though with not complaints whatsoever.

@arimendelow 

### Current workaround

Use relative imports and specify the full component path (i.e. `../src/components/ConfirmDialog/ConfirmDialog` works for the above example).


### How do we reproduce the bug?

1. Utilize cli-storybook-vite and edit `main.ts` or `preview.tsx` in `web/.storybook`
2. Try adding an absolute path to any component starting with `src/` *or* any path to a component that only refers to the component directory (i.e. `../src/components/MyComponent` instead of `../src/components/MyComponent/MyComponent`)

### What's your environment? (If it applies)

```shell
System:
    OS: Linux 6.12 Manjaro Linux
    Shell: 5.2.37 - /bin/bash
  Binaries:
    Node: 20.16.0 - /tmp/xfs-88cf9d03/node
    Yarn: 4.4.0 - /tmp/xfs-88cf9d03/yarn
  Databases:
    SQLite: 3.46.1 - /usr/bin/sqlite3
  npmPackages:
    @redwoodjs/auth-dbauth-setup: 8.4.1 => 8.4.1 
    @redwoodjs/cli-storybook-vite: 8.4.1 => 8.4.1 
    @redwoodjs/core: 8.4.1 => 8.4.1 
    @redwoodjs/project-config: 8.4.1 => 8.4.1
```


### Are you interested in working on this?

- [ ] I'm interested in working on this

--- a/.changesets/11746.md
+++ b/.changesets/11746.md
@@ -1,0 +1,3 @@
+- Fix absolute & component folder imports in Storybook config folder (#11746) by @Philzen
+
+Include the storybook config folder in the includes in `{ts,js}config.json`

--- a/__fixtures__/test-project/web/tsconfig.json
+++ b/__fixtures__/test-project/web/tsconfig.json
@@ -35,6 +35,7 @@
   "include": [
     "src",
     "config",
+    ".storybook/**/*",
     "../.redwood/types/includes/all-*",
     "../.redwood/types/includes/web-*",
     "../types",

--- a/packages/create-redwood-app/templates/js/web/jsconfig.json
+++ b/packages/create-redwood-app/templates/js/web/jsconfig.json
@@ -43,7 +43,7 @@
   },
   "include": [
     "src",
-    "config",
+    ".storybook/*",
     "../.redwood/types/includes/all-*",
     "../.redwood/types/includes/web-*",
     "../types",

--- a/packages/create-redwood-app/templates/js/web/jsconfig.json
+++ b/packages/create-redwood-app/templates/js/web/jsconfig.json
@@ -43,6 +43,7 @@
   },
   "include": [
     "src",
+    "config",
     ".storybook/*",
     "../.redwood/types/includes/all-*",
     "../.redwood/types/includes/web-*",

--- a/packages/create-redwood-app/templates/js/web/jsconfig.json
+++ b/packages/create-redwood-app/templates/js/web/jsconfig.json
@@ -44,7 +44,7 @@
   "include": [
     "src",
     "config",
-    ".storybook/*",
+    ".storybook/**/*",
     "../.redwood/types/includes/all-*",
     "../.redwood/types/includes/web-*",
     "../types",

--- a/packages/create-redwood-app/templates/ts/web/tsconfig.json
+++ b/packages/create-redwood-app/templates/ts/web/tsconfig.json
@@ -34,7 +34,7 @@
   },
   "include": [
     "src",
-    "config",
+    ".storybook/*",
     "../.redwood/types/includes/all-*",
     "../.redwood/types/includes/web-*",
     "../types",

--- a/packages/create-redwood-app/templates/ts/web/tsconfig.json
+++ b/packages/create-redwood-app/templates/ts/web/tsconfig.json
@@ -34,6 +34,7 @@
   },
   "include": [
     "src",
+    "config",
     ".storybook/*",
     "../.redwood/types/includes/all-*",
     "../.redwood/types/includes/web-*",

--- a/packages/create-redwood-app/templates/ts/web/tsconfig.json
+++ b/packages/create-redwood-app/templates/ts/web/tsconfig.json
@@ -35,7 +35,7 @@
   "include": [
     "src",
     "config",
-    ".storybook/*",
+    ".storybook/**/*",
     "../.redwood/types/includes/all-*",
     "../.redwood/types/includes/web-*",
     "../types",

